### PR TITLE
Adds Eth feeHistory & maxPriorityFeePerGas rpc

### PR DESCRIPTION
**Pull request by crystalin**

### CODE REVIEW

File: packages/types/src/interfaces/eth/rpc.ts [hunk 1]
```diff
@@ -178,6 +178,24 @@ export const rpc: DefinitionsRpc = objectSpread({}, netRpc, web3Rpc, {
...
+        name: "blockCount",
+        type: "U256"
+      },
+      {
+        name: "newestBlock",
+        type: "BlockNumber"
+      },
+      {
+        name: "rewardPercentiles",
+        type: "Option<Vec<f64>>"
```
Comment (jacogr) [inline packages/types/src/interfaces/eth/rpc.ts:194]: Need to look at this type - not sure we have an f64 inside the API. (I know the `f*` have been added recently as a hack-y thing to scale-codec)

**Comment by jacogr** [comment]
Just need to run `yarn lint` locally (it will auto-fix the `"`, etc)

File: packages/types/src/interfaces/eth/rpc.ts (hunk 1 above)
Comment (jacogr) [reply packages/types/src/interfaces/eth/rpc.ts:194]: Relevant scale-codec PR - https://github.com/paritytech/parity-scale-codec/pull/340

**Comment by crystalin** [comment]
Thank you @jacogr 
(too bad github integrated vscode doesn't allow to run yarn, it is such a convenient system to do quick modifications :)
I run `yarn lint` locally and pushed

**Comment by jacogr** [comment]
Seems spot-on, thank you. However, according to CI first need to get f64 support in before it is happy. (The PR is there, but will be a couple of days before it lands)

**Comment by crystalin** [comment]
Thank you @jacogr , there is no rush for this one

**Comment by jacogr** [comment]
Could you please merge in master? (f64 support is there, so it will make CI pass)

**Comment by crystalin** [comment]
@jacogr I merged master

**Comment by jacogr** [comment]
Ummm... weird... https://github.com/polkadot-js/api/runs/7004382425?check_suite_focus=true#step:3:186

That is certainly my issue, however I did check these in the API [via createType](https://github.com/polkadot-js/api/blob/master/packages/api/src/checkTypes.manual.ts#L216) and it was fine. Will debug, sorry about that. (Have really not added primitives in a while, so actually not overly surprised I missed a step... somewhere...)

**Comment by crystalin** [comment]
I was surprised by that too. I let you dig into it, if you want me to try again, let me know

**Comment by jacogr** [comment]
Please try again with merging master... it worked in my simple testes for typegen now.

**Comment by crystalin** [comment]
Done

**Comment by jacogr** [comment]
Small ordering issue in the rpc.ts file that CI is complaining about, but seems all ok now for the rest.

**Comment by crystalin** [comment]
@jacogr 
Should be good now, sorry for that.
Concerning lint, it retuns many error so I didn't know what to do:
```
yarn polkadot-exec-eslint --fix --resolve-plugins-relative-to /home/crystalin/projects/polkadotjs-api/node_modules/@polkadot/dev/scripts --ext .js,.cjs,.mjs,.ts,.tsx /home/crystalin/projects/polkadotjs-api
$ eslint --fix --resolve-plugins-relative-to /home/crystalin/projects/polkadotjs-api/node_modules/@polkadot/dev/scripts --ext .js,.cjs,.mjs,.ts,.tsx /home/crystalin/projects/polkadotjs-api
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting.

/home/crystalin/projects/polkadotjs-api/packages/api/src/base/Decorate.ts
  758:9  error  Unsafe return of an `any` typed value  @typescript-eslint/no-unsafe-return
  758:9  error  Unsafe call of an `any` typed value    @typescript-eslint/no-unsafe-call

/home/crystalin/projects/polkadotjs-api/packages/api/src/promise/decorateMethod.ts
  79:9  error  Unsafe call of an `any` typed value  @typescript-eslint/no-unsafe-call
  98:9  error  Unsafe call of an `any` typed value  @typescript-eslint/no-unsafe-call

/home/crystalin/projects/polkadotjs-api/packages/api/src/types/consts.ts
  4:15  error  No named exports found in module '@polkadot/api-base/types/consts'  import/export

/home/crystalin/projects/polkadotjs-api/packages/api/src/types/errors.ts
  4:15  error  No named exports found in module '@polkadot/api-base/types/errors'  import/export

/home/crystalin/projects/polkadotjs-api/packages/api/src/types/events.ts
  4:15  error  No named exports found in module '@polkadot/api-base/types/events'  import/export

/home/crystalin/projects/polkadotjs-api/packages/api/src/types/index.ts
  20:15  error  No named exports found in module '@polkadot/api-base/types'         import/export
  21:15  error  No named exports found in module '@polkadot/api/types/consts'       import/export
  22:15  error  No named exports found in module '@polkadot/api/types/errors'       import/export
  23:15  error  No named exports found in module '@polkadot/api/types/events'       import/export
  24:15  error  No named exports found in module '@polkadot/api/types/storage'      import/export
  25:15  error  No named exports found in module '@polkadot/api/types/submittable'  import/export

/home/crystalin/projects/polkadotjs-api/packages/api/src/types/storage.ts
  4:15  error  No named exports found in module '@polkadot/api-base/types/storage'  import/export

/home/crystalin/projects/polkadotjs-api/packages/api/src/types/submittable.ts
  4:15  error  No named exports found in module '@polkadot/api-base/types/submittable'  import/export

✖ 15 problems (15 errors, 0 warnings)

node:internal/errors:841
  const err = new Error(message);
```

**Comment by jacogr** [comment]
The above are weird - what is your TS version?

**Comment by crystalin** [comment]
> The above are weird - what is your TS version? (It needs at least 4.1+)

I don't have TS or ts-node globally installed, so it uses the ones from the yarn:
```
> node -v
v16.15.1
```

**Pull request merged**